docs/testrunner-toolkit/configuration/xcuitest.md

```diff
@@ -1,190 +1,468 @@
+```yaml
+  rootDir: "./" # Use the current directory
+```
 
-When an ID is specified, it supersedes the other settings.
+```yaml
+  rootDir: "packages/subpackage" # Some other package from within a monorepo
+```
+---
+
+## `artifacts`
+<p><small>| OPTIONAL | OBJECT |</small></p>
 
-__Type__: *[]object*
+Specifies how to manage test output, such as logs, videos, and screenshots.
 
-__Example__:
 ```yaml
-devices:
-  - name: "iPhone 11"
-    platformVersion: "14.3"
-  - id: iPhone_11_14_5_real_us
+artifacts:
+  download:
+    when: always
+    match:
+      - junit.xml
+    directory: ./artifacts/
 ```
+---
 
-### `id`
+### `download`
+<p><small>| OPTIONAL | OBJECT |</small></p>
 
-__Description__: Request a specific device by its id. You can look up device IDs in the Sauce Labs app or using our [Get Devices API request](https://docs.saucelabs.com/dev/api/rdc#get-devices).
+Specifies the settings related to downloading artifacts from tests run by `saucectl`.
 
-__Type__:  *string*
+```yaml
+  download:
+    when: always
+    match:
+      - junit.xml
+    directory: ./artifacts/
+```
+---
+
+#### `when`
+<p><small>| OPTIONAL | STRING |</small></p>
 
-__Example__:
+Specifies when and under what circumstances to download artifacts. Valid values are:
+
+* `always`: Always download artifacts.
+* `never`: Never download artifacts.
+* `pass`: Download artifacts for passing suites only.
+* `fail`: Download artifacts for failed suites only.
 
 ```yaml
-        id: iPhone_11_14_5_real_us
+    when: always
 ```
+---
 
-### `name`
+#### `match`
+<p><small>| OPTIONAL | STRING/ARRAY |</small></p>
 
-__Description__:  Request a device by its name.
+Specifies which artifacts to download based on whether they match the name or file type pattern provided. Supports the wildcard character `*` so you can conveniently specify all artifacts of a specific file type.
 
-__Type__:  *string*
+```yaml
+  match:
+    - junit.xml
+    - *.log
+```
+---
 
-__Example__:
+#### `directory`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+Specifies the path to the folder location in which to download artifacts. A separate subdirectory is generated in this location for each suite for which artifacts are downloaded.
 
-Using complete name:
 ```yaml
-      - name: iPhone 11
+    directory: ./artifacts/
 ```
+---
+
+## `xxcuitest`
+<p><small>| REQUIRED | OBJECT |</small></p>
+
+The parent property containing the details specific to the XCUITest project.
 
-Using pattern matching:
 ```yaml
-        name: iPhone*
+xcuitest:
+  app: ./apps/SauceLabs.Mobile.Sample.XCUITest.App.ipa
+  testApp: ./apps/SwagLabsMobileAppUITests-Runner.app
 ```
+---
+
+### `app`
+<p><small>| REQUIRED | STRING |</small></p>
 
-### `platformVersion`
+The directory path to the application. The default directory is `{project-root}/apps/filename.ipa`, and the property supports expanded environment variables to designate the path, as shown in the following examples.
```

Comment:
We don't use any default value for that field. 

I think  `The directory path to the application.`  should be rephrased as `The path to the application.` as XCUITest applications can be both a file or a folder.

Basically, if it's an `.ipa` it needs to be a file, if it's an `.app` it needs to be a directory.